src/compiler/moduleSpecifiers.ts

```diff
@@ -449,6 +485,25 @@ namespace ts.moduleSpecifiers {
                 else if (pattern === relativeToBaseUrl || pattern === relativeToBaseUrlWithIndex) {
                     return key;
                 }
+                else if (host?.readFile && host.fileExists(combinePaths(pathsBasePath!, pattern, "package.json"))) {
+                    // If the path points to a directory with a package.json, see if we can resolve to its 'main' field.
+                    try {
+                        const { main } = JSON.parse(host.readFile(combinePaths(pathsBasePath!, pattern, "package.json"))!);

```

Comment:
why just main.. I think during resoution we also look for types and typings field and that is preferred over main ?